ipalib/rpc.py

```diff
@@ -118,7 +119,10 @@ def read_persistent_client_session_data(principal):
     '''
 
     try:
-        return session_storage.get_data(principal, CCACHE_COOKIE_KEY)
+        value = session_storage.get_data(principal, CCACHE_COOKIE_KEY)
+        if value == CCACHE_COOKIE_EMPTY_VALUE:
```

Comment:
Please move the three lines out of the try/except block. It doesn't make sense to catch and reraise the error.

Under Python 3, does get_data return string or bytes?